### PR TITLE
[ffigen] Use package:dart_style

### LINKED ISSUE #1664: [ffigen] Use package:dart_style
Use package:dart_style directly rather than subprocessing to `dart format`. Should be faster and more reliable. Also would let us remove a g3 patch (actually we can pretty much just apply g3/third_party/dart/ffigen/patches/dart_format.diff to this repo).

### PR DESCRIPTION
**Pull request by liamappelbe**
Use `package:dart_style` directly to format generated Dart code, rather than subprocessing to `dart format`.

Fixes #1664

### CODE REVIEW

**Review by HosseinYousefi — APPROVED** [review]
LGTM! CI is failing though, maybe some coverage related thing again.

**Comment by liamappelbe** [comment]
> LGTM! CI is failing though, maybe some coverage related thing again.

Yeah, another bug I can't repro locally. This PR is low priority, so I'll fix the coverage issue first.

**Comment by liamappelbe** [comment]
The fix for the coverage failure is in https://github.com/dart-lang/tools/pull/1189. I'll wait for that to land and pull in the new coverage version before landing this.

**Pull request merged**
